### PR TITLE
Feat: add gemma3n support

### LINKED ISSUE #2850: Adding Gemma 3n support?
### ⚠️ Please check that this feature request hasn't been suggested before.

- [x] I searched previous [Ideas in Discussions](https://github.com/axolotl-ai-cloud/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [x] I searched previous [Issues](https://github.com/axolotl-ai-cloud/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

I saw you guys were mentioned in the [Gemma 3n Developer's Guide](https://developers.googleblog.com/en/introducing-gemma-3n-developer-guide/) and wanted to check if you were tracking to add support for the [family of MatFormer models](https://huggingface.co/collections/google/gemma-3n-685065323f5984ef315c93f4) recently released by Google.

### ✔️ Solution

I tried using the [gemma3 examples](https://github.com/axolotl-ai-cloud/axolotl/tree/main/examples/gemma3) as a starting point, and after upgrading transformers to [v4.53.0](https://github.com/huggingface/transformers/releases/tag/v4.53.0) and installing timm, I still got opaque errors like 

> ) is not supported. Currently, only the following modules are supported: `torch.nn.Linear`, `torch.nn.Embedding`, `torch.nn.Conv1d`, `torch.nn.Conv2d`, `torch.nn.Conv3d`, `transformers.pytorch_utils.Conv1D`, `torch.nn.MultiheadAttention.`.

I'd love it if, like most other models I've finetuned with axolotl, it just worked.

### ❓ Alternatives

Unsloth says they have [Gemma 3n support](https://docs.unsloth.ai/basics/gemma-3n-how-to-run-and-fine-tune), but their page just says "Gemma 3n Notebooks coming soon! Stay tuned."

### 📝 Additional Context

_No response_

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this feature has not been requested yet.
- [x] I have provided enough information for the maintainers to understand and evaluate this request.

### PR DESCRIPTION
**Pull request by NanoCode012**
# Description

Requires `pip install timm`

TODO:
- [x] add gemma3n chat template

Limitations:
- The loss is abnormal even training a small subset of models
- The VRAM usage is high (likely due to hidden states in fp32)

This PR also adds audio training, fixes vision, and minor fix to prompt formatting.

## Summary by CodeRabbit

* **New Features**
  * Added support for the Gemma-3n model, including new processing strategies and a dedicated chat template.
  * Introduced configuration files for Gemma-3n fine-tuning with text, vision, and audio modalities.
  * Extended dataset format to support audio content.
  * Added documentation and usage examples for Gemma-3n, including installation tips for required libraries.

* **Bug Fixes**
  * Improved error messaging and added FAQ for image loading issues.

* **Documentation**
  * Updated multimodal documentation and Cut Cross Entropy integration to reflect new model support and dataset changes.

* **Refactor**
  * Adjusted batch collation to support additional multi-modal input features and attention masks.

* **Style**
  * Standardized optimizer and attention mechanism settings across example configuration files.

### CODE REVIEW

File: examples/gemma3n/gemma-3n-e2b-vision-audio-qlora.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+chat_template: gemma3n
+eot_tokens:
+  - <end_of_turn>
+
+datasets:
+  - path: Nanobit/text-vision-audio-2k-test  # requires downloading audio/image in advance in README.md
+    type: chat_template
+    data_files:
+      - dataset.jsonl
```
Comment (NanoCode012) [inline examples/gemma3n/gemma-3n-e2b-vision-audio-qlora.yml]: I think HF is buggy with uploading audio/image, so this ds config requires downloading the raw files in advance.

**Comment by salmanmohammadi** [comment]
> pip install timm && pip install -U transformers

Is this still the case? @NanoCode012

**Comment by NanoCode012** [comment]
> > pip install timm && pip install -U transformers
> 
> Is this still the case? @NanoCode012

`timm` only now

**Review by salmanmohammadi — APPROVED** [review]
